libs/dh/reports/feature-measurement-reports/src/components/request-report/request-as-modal.component.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  FormControl,
+  FormGroup,
+  NonNullableFormBuilder,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { RxPush } from '@rx-angular/template/push';
+import { TranslocoDirective } from '@jsverse/transloco';
+import { Apollo } from 'apollo-angular';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { WattButtonComponent } from '@energinet-datahub/watt/button';
+import { WattDropdownComponent } from '@energinet-datahub/watt/dropdown';
+import { WATT_MODAL, WattModalService, WattTypedModal } from '@energinet-datahub/watt/modal';
+import { VaterStackComponent } from '@energinet-datahub/watt/vater';
+import { getActorOptions } from '@energinet-datahub/dh/shared/data-access-graphql';
+import { EicFunction, GetActorByIdDocument } from '@energinet-datahub/dh/shared/domain/graphql';
+import {
+  DhActorStorage,
+  PermissionService,
+} from '@energinet-datahub/dh/shared/feature-authorization';
+
+import { DhRequestReportModal } from './request-report-modal.component';
+
+type DhFormType = FormGroup<{
+  actorId: FormControl<string>;
+}>;
+
+@Component({
+  selector: 'dh-request-as-modal',
+  imports: [
+    RxPush,
+    ReactiveFormsModule,
+    TranslocoDirective,
+
+    WATT_MODAL,
+    VaterStackComponent,
+    WattDropdownComponent,
+    WattButtonComponent,
+  ],
+  templateUrl: './request-as-modal.component.html',
+})
+// eslint-disable-next-line @angular-eslint/component-class-suffix
+export class DhRequestAsModal extends WattTypedModal {
+  private readonly formBuilder = inject(NonNullableFormBuilder);
+  private readonly modalService = inject(WattModalService);
+  private readonly destroyRef = inject(DestroyRef);
+  private readonly apollo = inject(Apollo);
+
+  private readonly permissionService = inject(PermissionService);
+  private readonly actorStorage = inject(DhActorStorage);
+
+  form: DhFormType = this.formBuilder.group({
+    actorId: new FormControl<string>(this.actorStorage.getSelectedActorId(), {
+      validators: Validators.required,
+      nonNullable: true,
+    }),
+  });
+
+  actorOptions$ = getActorOptions(
+    [EicFunction.DataHubAdministrator, EicFunction.GridAccessProvider],
+    'actorId'
+  );
+
+  submitInProgress = signal(false);
+
+  submit(): void {
+    if (this.form.invalid || !this.form.value.actorId) {
+      return;
+    }
+
+    this.submitInProgress.set(true);
+
+    if (this.form.value.actorId == this.actorStorage.getSelectedActorId()) {
+      this.permissionService
+        .isFas()
+        .pipe(takeUntilDestroyed(this.destroyRef))
+        .subscribe((isFas) => {
+          this.modalService.close(true);
+          this.modalService.open({
+            component: DhRequestReportModal,
+            data: {
+              isFas,
+              actorId: this.actorStorage.getSelectedActor()?.id,
+              marketRole: this.actorStorage.getSelectedActor()?.marketRole,
+            },
+          });
+
+          this.submitInProgress.set(false);
```

Comment:
Suggestion: You could move the open modal logic into a function, and reuse

Reply:
Did you mean something like [this](https://github.com/Energinet-DataHub/greenforce-frontend/pull/4316/commits/6c2c1e54b96a05e2acdfe6d2d1bc920139c0e867)